src/autoformatengine.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Range from '../engine/model/range.js';
+import TextProxy from '../engine/model/textproxy.js';
+
+export default class AutoformatEngine {
```

Comment:
This class is just a constructor injecting some behavior using parameters passed to that constructor. We could turn this whole class into a simple function.